### ISSUE TITLE
Problem with frege.java.Util.Properties

### ISSUE THREAD

**Issue by ForNeVeR** [Frege/frege#129]
Some methods of `Properties` type seems to be incorrectly typed. For example, consider `load` method:

```frege
data Properties = native java.util.Properties where
  ...
  native load :: MutableIO Properties -> MutableIO Reader -> IO () throws IOException
               | MutableIO Properties -> MutableIO InputStream -> IO () throws IOException

```

It accepts only `MutableIO InputStream` but the language doesn't provide an ability to get `MutableIO InputStream` at all.

The same seems to be case with almost all methods of `Properties` type. I've discussed it with @Ingo60 in IRC and he thinks it may be caused due to problem with `frege-native-gen` (I'll check it eventually), so we maybe should rise another issue there.

The solution may be to simply remove `MutableIO` from `InputStream` (and maybe some another types) in method signatures of `Properties`. If the `Util` class was auto-generated and meant to be changed only by regeneration - we should fix the `frege-native-gen` first.

I am ready to provide some code samples about which I think they should compile and they are currently not.

**Comment by mmhelloworld** [comment]
Isn't InputStream substitutable for `MutableIO Inputstream` as it is a
`mutable native` type?

**Comment by ForNeVeR** [comment]
As i can seen - no, it don't. Citing the documentation:

> A native type _M_ whose values would, by the rules above, appear _everywhere_ with type
`Mutable RealWorld M`, is called a _mutable only_ type. It can be declared like
> 
> ```frege
data M = mutable native ...
```
> 
> and can appear in type signatures as just _M_. The compiler will then apply the rules
above _as if_ the type was `Mutable RealWorld M`, [...]

So it is not a `MutableIO` (remember that `MutableIO = Mutable RealWorld`).

**Comment by Ingo60** [comment]
No, it isn't. Since a type constructor can never unify with a type
application. I suggest editing Util.fr with find/replace for the critical
types.

**Comment by mmhelloworld** [comment]
I think I just read this `The compiler will then apply the rules
above as if the type was Mutable RealWorld M` and misunderstood. 

Now it is clear from this statement from language reference `The type checker expects a Mutable s m,
but the 'as if' above concerns only sanity checks for types of native functions, the
normal typing rules will never actually unify Mutable s m with M .`

I will send a pull request later today to fix this.

**Comment by Ingo60** [comment]
I shall also let the compiler emit a warning when it sees such "impossible" types, for easier diagnosis.

**Comment by mmhelloworld** [comment]
That would be better! Thanks! 

Here is the pull request to fix Util.fr: https://github.com/Frege/frege/pull/130

**Comment by Ingo60** [comment]
Reopened to keep track of compiler changes.

**Comment by Ingo60** [comment]
Fixed with commit 31d2574d471f1754fd88353f0c1eac2e92d315c0

If a type is mutable only, it must not appear wrapped in Mutable.
